src/daemon/abrt-action-save-container-data.c

```diff
@@ -29,58 +29,109 @@ void dump_docker_info(struct dump_dir *dd, const char *root_dir)
     FILE *mntnf_file = fopen(mntnf_path, "r");
     free(mntnf_path);
 
-    struct mountinfo mntnf;
-    int r = get_mountinfo_for_mount_point(mntnf_file, &mntnf, "/");
-    fclose(mntnf_file);
+    struct mount_point {
+        const char *name;
+        enum mountinfo_fields {
+            MOUNTINFO_ROOT,
+            MOUNTINFO_SOURCE,
+        } field;
+    } mount_points[] = {
+        { "/sys/fs/cgroup/memory", MOUNTINFO_ROOT },
+        { "/",                     MOUNTINFO_SOURCE },
+    };
 
-    if (r != 0)
-    {
-        error_msg("dockerized processes must have re-mounted root");
-        goto dump_docker_info_cleanup;
-    }
+    char *container_id = NULL;
+    char *output = NULL;
 
-    const char *mnt_src = MOUNTINFO_MOUNT_SOURCE(mntnf);
-    const char *last = strrchr(mnt_src, '/');
-    if (last == NULL || strncmp("/docker-", last, strlen("/docker-")) != 0)
-    {
-        error_msg("Mounted source is not a docker mount source");
-        goto dump_docker_info_cleanup;
-    }
+    /* initialized to 0 because we call mountinfo_destroy below */
+    struct mountinfo mntnf = {0};
 
-    last = strrchr(last, '-');
-    if (last == NULL)
+    for (size_t i = 0; i < ARRAY_SIZE(mount_points); ++i)
     {
-        error_msg("The docker mount source has unknown format");
-        goto dump_docker_info_cleanup;
+        log_debug("Parsing container ID from mount point '%s'", mount_points[i].name);
+
+        rewind(mntnf_file);
+
+        /* get_mountinfo_for_mount_point() re-initializes &mntnf */
+        mountinfo_destroy(&mntnf);
+        int r = get_mountinfo_for_mount_point(mntnf_file, &mntnf, mount_points[i].name);
+
+        if (r != 0)
+        {
+            log_debug("Mount poin not found");
+            continue;
+        }
+
+        const char *mnt_info = NULL;
+        switch(mount_points[i].field)
+        {
+            case MOUNTINFO_ROOT:
+                mnt_info = MOUNTINFO_ROOT(mntnf);
+                break;
+            case MOUNTINFO_SOURCE:
+                mnt_info = MOUNTINFO_MOUNT_SOURCE(mntnf);
+                break;
+            default:
+                error_msg("BUG: forgotten MOUNTINFO field type");
+                abort();
+        }
+        const char *last = strrchr(mnt_info, '/');
+        if (last == NULL || strncmp("/docker-", last, strlen("/docker-")) != 0)
+        {
+            log_debug("Mounted source is not a docker mount source: '%s'", mnt_info);
+            continue;
+        }
+
+        last = strrchr(last, '-');
+        if (last == NULL)
+        {
+            log_debug("The docker mount point has unknown format");
+            continue;
+        }
+
+        ++last;
+        container_id = xstrndup(last, 12);
```

Comment:
Sorry, I don't see any reason for it.  The value is literally a magic constant. I will add a comment.